backend/routes/userRoutes.js

```diff
@@ -105,4 +105,142 @@ router.patch('/location', async(req, res) => {
+  )
+  let numberOfEventsAttending = Object.keys(eventsAttending).length
+  let meanDistanceTraveledUser = totalDistanceUserToEvent/numberOfEventsAttending
+  // -------------------------------------------------------------------------------
+  // Get the events that are nearby (square with side = 28 miles)
+  const eventsNear = await prisma.event.findMany({
+    where: { 
+      lat: {
+        gt: user.lat - 0.2,
+        lt: user.lat + 0.2
+      },
+      long: {
+        gt: user.long - 0.2, 
+        lt: user.long + 0.2 
+      }
+    }
+  })
+  // ------------------------------------------------
+  // Get current date (https://stackoverflow.com/questions/1531093/how-do-i-get-the-current-date-in-javascript)
+  var today = new Date();
+  // ---------------------------------------------------------------------------
+  // Calculate the score for each event ****************************************************************************************
+  let eventDate = new Date();
+  let dateDifference = 0
+  let dateScore = 0
+  let countOfApperances = 0
+  let categoryScore = 0
+  let distanceEventToUser = 0
+  let distanceScore = 0
+  let totalScore = 0
+
+  eventsNear.map((e) => {
+    // Date Score -----------------------------------------------------------------------------------------------------------
+    eventDate = new Date(parseInt(e.date.slice(0, 2)) + '/' + parseInt(e.date.slice(3, 5)) + '/' + parseInt(e.date.slice(6, 10)))
+    dateDifference =  ( ((eventDate.getTime() - today.getTime()) / 1000) / 604800 ) 
+
+    if (dateDifference !=0 ){
+      dateScore = 1 / dateDifference
+    } else{
+      dateScore = 1
+    }
+    
+    // ------------------------------------------------------------------------------------------------------------------------
+    // Category Score -----------------------------------------------------------------------------------------------------------
+    countOfApperances = 0
+    for (let i = 0; i < categoriesEventsAttending.length; i++) {
+      if (categoriesEventsAttending[i] == e.category){
+        countOfApperances = countOfApperances + 1
+      }
+    }
+    if (categoriesEventsAttending.length != 0) {
+      categoryScore = (countOfApperances)/(categoriesEventsAttending.length)
+    } else {
+      categoryScore = 0
+    }
+    // ------------------------------------------------------------------------------------------------------------------------
+    // Distance Score -----------------------------------------------------------------------------------------------------------
+    distanceEventToUser = ( ((e.lat - user.lat)**2 + (e.long - user.long)**2)**.5 )
+
+    if ((meanDistanceTraveledUser - distanceEventToUser) > 0){
+      distanceScore = 1 / (meanDistanceTraveledUser - distanceEventToUser)
+    } else if ((meanDistanceTraveledUser - distanceEventToUser) < 0) {
+      distanceScore = 1 / (distanceEventToUser - meanDistanceTraveledUser)
+    }
+    else {
+      distanceScore = 1
+    }
```

Comment:
From this algorithm, the closer the event is to the mean distance user traveled is better. However, in reality, wouldn't the closer the event is to my home the better?

Reply:
Solved in PR 39 by averaging both